[projections] Fix performance regression in draw_background

mourner opened the pull request:
An omission caused `draw_background` to regenerate raster/stencil buffers on every frame instead of reusing ones from cached tiles. This is a quick fix so that it takes any available source cache from the style and uses that for the background tiles, removing the bottleneck. It's likely not passing Flow tests now but we can fix that after consolidating with @ryanhamley's branch.

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] manually test the debug page

ansis reviewed (changes requested):
It looks like this only renders the background for tiles that have loaded when it should render it for the entire viewport. I guess it could render all tiles from that source, not just "renderable" ones, but it might be cleaner to have a separate sourcecache just for the background.

mourner commented:
> it might be cleaner to have a separate sourcecache just for the background

@ansis it might be cleaner but appears to be a pretty difficult refactor because of the way `SourceCache` wraps around a certain `Source` (which a background layer wouldn't have), unless you're seeing an easy way to implement this? It might be fine to have background only rendered for loaded tiles and create a ticket for cleaning this up later, because it's a much smaller issue than the performance bottleneck.

ansis commented:
I was imagining something like https://github.com/mapbox/mapbox-gl-js/tree/projections-fix-bg-perf-suggestion

Do you think this would work well enough?

mourner commented:
@ansis haven't thought of this — very neat fix! Will rebase this against the `projections` branch and force-push.

mourner commented:
@ansis Rebased, cleaned up and fixed flow / lint. It's still not a perfect solution because we basically duplicate the work, which is still quite significant according to the profiler. Ideally we would refactor things so that the tile pyramid / tile buffers are separate from tile data so that the first wouldn't be duplicated, but we can try doing that later.

![image](https://user-images.githubusercontent.com/25395/122754953-f6aa0200-d29c-11eb-8a21-c53603ce58a4.png)

mourner commented:
Really confused by the render tests failures here — `test-render-prod` passes fully while `test-render` fails 256 tests for unclear reasons...

ryanhamley commented:
I re-ran the render tests and they passed. I've noticed some flakiness around individual tests in the projections work, but nothing where hundreds of tests unexpectedly failed. I wonder if there was Circle cached an old version of GL JS or something weird like that.

1z2x3c4v5b6n7m8 commented:
***@***.***
